File: gamepack/Decompiled/SpriteMask.java
public class SpriteMask extends DualNode {
   static String gn;
   public final int width;
   public final int height;
   public final int[] xStarts;
   public final int[] xWidths;

   public static int af_renamed(int var0, int var1, int var2) {
      var2 &= 3;
      if (var2 == 0) {
         return var0;
      } else if (1 == var2) {
         return var1;
      } else {
         return 2 == var2 ? 7 - var0 : 7 - var1;
      }
   }

   SpriteMask(int var1, int var2, int[] var3, int[] var4) {
      this.width = var1 * 1531475491;
      this.height = var2 * 1387260015;
      this.xWidths = var3;
      this.xStarts = var4;
   }

   public boolean contains(int var1, int var2) {
      if (var2 >= 0 && var2 < this.xStarts.length) {
         int var4 = this.xStarts[var2];
         if (var1 >= var4 && var1 <= var4 + this.xWidths[var2]) {
            return true;
         }
      }

      return false;
   }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/classes/SpriteMask.kt
package org.runestar.client.updater.mapper.identifiers.classes

import org.objectweb.asm.Type.BOOLEAN_TYPE
import org.objectweb.asm.Type.INT_TYPE
import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.annotations.DependsOn
import org.runestar.client.updater.mapper.annotations.MethodParameters
import org.runestar.client.updater.mapper.identifiers.SpriteMaskConstructorField
import org.runestar.client.updater.mapper.predicateutilities.and
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.predicateutilities.type
import org.runestar.client.updater.mapper.wrappers.Class2
import org.runestar.client.updater.mapper.wrappers.Method2

@DependsOn(DualNode::class)
class SpriteMask : IdentityMapper.Class() {

    override val predicate = predicateOf<Class2> { it.superType == type<DualNode>() }
            .and { it.instanceFields.count { it.type == IntArray::class.type } == 2 }
            .and { it.instanceFields.count { it.type == INT_TYPE } == 2 }
            .and { it.instanceMethods.size == 1 }
    class width : SpriteMaskConstructorField(INT_TYPE, 0)
    class height : SpriteMaskConstructorField(INT_TYPE, 1)
    class xWidths : SpriteMaskConstructorField(IntArray::class.type, 0)
    class xStarts : SpriteMaskConstructorField(IntArray::class.type, 1)

    @MethodParameters("x", "y")
    class contains : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<Method2> { it.returnType == BOOLEAN_TYPE }
    }
}